test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DateTimeSupport/DateTimeTest.cs

```diff
@@ -83,6 +84,35 @@ public async Task ModelBuilderTest(string modelMode)
             Assert.False(modifiedDates.Type.IsNullable);
         }
 
+        [Theory]
+        [InlineData("convention")]
+        [InlineData("explicit")]
+        public async Task PostToDateTimeCollection(string modelMode)
+        {
+            await ResetDatasource();
+            //Arrange
+            string requestUri = string.Format("{0}/{1}/Files/1/ModifiedDates", this.BaseAddress, modelMode);
+            var requestForPost = new HttpRequestMessage(HttpMethod.Post, requestUri);
+
+            requestForPost.Content = new StringContent(content: @"{
+                    'value':'2014-10-16T01:02:03Z'
+                    }", encoding: Encoding.UTF8, mediaType: "application/json");
+
+            using (HttpResponseMessage response = await this.Client.SendAsync(requestForPost))
+            {
+                Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+            }
+
+            using (HttpResponseMessage response = await this.Client.GetAsync(requestUri))
+            {
+                response.EnsureSuccessStatusCode();
+
+                var json = await response.Content.ReadAsObject<JObject>();
+                var result = json.GetValue("value") as JArray;
+                Assert.Equal(4,result.Count);
+            }
```

Comment:
if you want to test the count, i think you can use "$count" segment. For example:  ~/Fiels/1/ModifiedDates/$count

Reply:
Add the count check before the post as well.

---
In reply to: [223518830](https://github.com/OData/webapi/pull/1628#discussion_r223518830) [](ancestors = 223518830)

Reply:
As above; he's checking the count of the response from the POST request to make sure it includes the added member, which is good.

---
In reply to: [223518830](https://github.com/OData/WebApi/pull/1628#discussion_r223518830) [](ancestors = 223518830)

Reply:
Added a request for the pre-condition.